test/rest/orders/preview_order_test.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:io';
+
+import 'package:coinbase_cloud_advanced_trade_client/src/models/credential.dart';
+import 'package:coinbase_cloud_advanced_trade_client/src/models/orders/order_side.dart';
+import 'package:coinbase_cloud_advanced_trade_client/src/models/orders/preview_order.dart';
+import 'package:coinbase_cloud_advanced_trade_client/src/rest/orders/orders.dart';
+import 'package:http/http.dart' as http;
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../../tools.dart';
+import 'preview_order_test.mocks.dart';
+import 'package:logging/logging.dart';
+
+import '../../test_constants.dart';
+
+@GenerateMocks([http.Client])
+void main() {
+  Logger.root.level = Level.ALL;
+  Logger.root.onRecord.listen((record) {
+    print('${record.level.name}: ${record.time}: ${record.message}');
+  });
+  group('Preview Order', () {
+    test('previews a market order', () async {
+      final mockClient = MockClient();
+      final successResponse = await getJsonFromFile(
+          'mocks/rest/orders/preview_order_success.json');
+
+      when(mockClient.post(any,
+              headers: anyNamed('headers'), body: anyNamed('body')))
+          .thenAnswer((_) async => http.Response(successResponse, 200));
+
+      final PreviewOrderResponse? previewOrderResponse = await previewOrder(
+          client: mockClient,
+          credential: credentials,
+          productId: 'BTC-USD',
+          side: OrderSide.buy,
+          orderConfiguration: {
+            'market_market_ioc': {'quote_size': '10'}
+          });
+
+      expect(previewOrderResponse, isNotNull);
+      expect(previewOrderResponse!.orderTotal, '10.00');
+      expect(previewOrderResponse.commissionTotal, '0.05');
+      expect(previewOrderResponse.quoteSize, 10.00);
+      expect(previewOrderResponse.baseSize, 0.001);
+      expect(previewOrderResponse.bestBid, '9999.00');
+      expect(previewOrderResponse.bestAsk, '10001.00');
+      expect(previewOrderResponse.previewId, 'PREVIEW-ID-12345');
+    });
+  });
+
+  group('Preview Order Integration', () {
+    test('previews a market order', () async {
+      final PreviewOrderResponse? previewOrderResponse = await previewOrder(
+          credential: credentials,
+          productId: 'BTC-USD',
+          side: OrderSide.buy,
+          orderConfiguration: {
+            'market_market_ioc': {'quote_size': '10'}
+          });
+
+      print(previewOrderResponse.toString());
```

Comment:
Should use a test logger instead of print. See other tests for an example of how this is implemented properly.